### PR TITLE
Removed GUI-specific video player code from the core package

### PR DESCRIPTION
**Pull request by devos50**
Since showing a video is the responsibility of the new GUI, the video player code should be removed from the `Core` package. Moreover, the video server should be started as soon as Tribler starts (at least, if it's enabled in the configuration).

In this PR, I removed the video server and VLC wrapper. Any functionality that the video server needs, is copied into the video server class.

I've tested the video server and it seems to work fine.

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@lfdversluis ready for your review. I tried to solve as much pylint/coverage errors as I could, however, some of these are hard to solve without more refactoring (for instance, by using deferreds in the video server).

File: Tribler/Core/APIImplementation/LaunchManyCore.py [hunk 1]
```diff
@@ -698,9 +699,9 @@ def early_shutdown(self):
             yield self.rtorrent_handler.shutdown()
         self.rtorrent_handler = None
 
-        if self.videoplayer:
-            yield self.videoplayer.shutdown()
-        self.videoplayer = None
+        if self.video_server:
+            yield self.video_server.shutdown_server()
```
Comment (lfdversluis) [inline Tribler/Core/APIImplementation/LaunchManyCore.py]: I do not see an inlineCallback above the function?

File: Tribler/Core/SessionConfig.py [hunk 2]
```diff
@@ -612,13 +614,13 @@ def get_dispersy_port(self):
     #
     # Config for swift tunneling e.g. dispersy traffic
     #
-    def get_videoplayer(self):
+    def get_videoserver(self):
```
Comment (lfdversluis) [inline Tribler/Core/SessionConfig.py]: I would assume this returned the videoserver object itself judging by the name.

File: Tribler/Core/SessionConfig.py [hunk 3]
```diff
@@ -612,13 +614,13 @@ def get_dispersy_port(self):
...
         """ Enable or disable VOD functionality (default = True).
         @param value Boolean.
         """
         return self.sessconfig.get(u'video', u'enabled')
 
-    def set_videoplayer(self, value):
+    def set_videoserver(self, value):
```
Comment (lfdversluis) [inline Tribler/Core/SessionConfig.py]: idem but setting

File: Tribler/Core/Video/VideoServer.py [hunk 4]
```diff
@@ -33,6 +39,47 @@ def __init__(self, port, session, video_player):
...
+                vi_dict['stream'][0].close()
+
+        self.vod_download = new_download
+
+    def get_vod_stream(self, dl_hash, wait=False):
+        if 'stream' not in self.vod_info[dl_hash] and self.session.get_download(dl_hash):
+            download = self.session.get_download(dl_hash)
+            vod_filename = self.get_vod_destination(download)
+            while wait and not os.path.exists(vod_filename):
+                time.sleep(1)
```
Comment (lfdversluis) [inline Tribler/Core/Video/VideoServer.py]: no sleeps in tests

**Comment by lfdversluis** [comment]
Comments given.

File: Tribler/Core/Video/VideoServer.py (hunk 4 above)
Comment (devos50) [inline Tribler/Core/Video/VideoServer.py]: This code is moved from `VideoPlayer.py` and I did not adjust the logic. I can refactor the video player to use deferreds but I don't want to do too much at once.

**Comment by devos50** [comment]
Processed/commented on comments of @lfdversluis, @whirm can you review?

**Comment by whirm** [comment]
You need to increment  SESSDEFAULTS_VERSION  when changing the settings

**Comment by devos50** [comment]
@whirm I didn't change the structure of the configuration file itself, only the name of some getters and setters in the class.

**Comment by whirm** [comment]
You changed the default value of an option, I would increase the number and add the change to the log

**Comment by devos50** [comment]
@whirm ready again

**Pull request merged**